.github/workflows/optimize-ci.yml

```diff
@@ -301,13 +300,14 @@ jobs:
   deploy-artifacts:
     if: github.event_name == 'push' && (github.ref_name == 'main' || startsWith(github.ref_name, 'stable/'))
     name: Deploy Artifacts
-    needs: [integration-tests]
+    needs: [ integration-tests, upgrade ]
     uses: ./.github/workflows/optimize-deploy-artifacts.yml
     secrets: inherit
 
   # Rerun failed jobs running on self-hosted runners in case of network issues or node preemption
   rerun-failed-jobs:
     needs:
+      - upgrade
```

Comment:
the warning here looks legit too. This should not be here if we end up having this on the unified status checks

Reply:
Yup overlooked to remove